File: Lab6.c
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

void fromLeft(char* text, int n) {
    int index = -1;
    for (int i = 0; i < n; i++) {
        if (text[i] == 'a') {
            index = i;
            break;
        }
    }
    printf("Your index from left: %d\n", index);
}

void fromRight(char* text, int n) {
    int index = -1;
    for (int i = n-1; i >= 0; i--) {
        if (text[i] == 'a') {
            index = i;
            break;
        }
    }
    printf("Your index from right: %d\n", index);
}

void palindrome(char* text, int n) {
    int bool = 1;
    for (int i = 0; i < floor(n/2); i++) {
        if (text[i] != text[n-1-i]) {
            bool = 0;
            break;
        }
    }
    printf("Mirror valid: %d\n", bool);
}

int countChar(char* text, int n, char a) {
    int count = 0;
    for (int i = 0; i < n; i++) 
        if (text[i] == a)
            count++;
    return count;
}

int countVowels(char* text, int n) {
    int count = 0;
    for (int i = 0; i < n; i++) {
        text[i] = tolower(text[i]); 
        if (text[i] == 'a' || text[i] == 'e' || text[i] == 'i' || text[i] == 'o' || text[i] == 'u')
            count++;
    }
    return count;
}

int main() {
    char text[30];
    printf("Enter massage: ");
    scanf("%s", text);
    char a;
    printf("Enter char: ");
    scanf(" %c", &a);//skip \n
    int n = strlen(text);
    fromLeft(text, n);
    fromRight(text, n);
    palindrome(text, n);
    printf("Char: %d\n", countChar(text, n, a));
    printf("Vowels: %d\n", countVowels(text, n));
    return 0;
}

File: Lab6.py
#from left

text = input("Enter massage:")
n = len(text)
index = 0
for i in range(n):
    if text[i] == 'a':
        index = i
        break
print(f"Your index: {index}")


#from right

text = input("Enter massage:")
n = len(text)
index = 0
for i in range(n-1, -1, -1):
    if text[i] == 'a':
        index = i
        break
print(f"Your index: {index}")


#palindrome
text = input("Enter massage:")
n = len(text)
bool = True
for i in range(n//2):
    if text[i] != text[n-1-i]:
        bool = False
        break
print(bool)


#count character (want only)

def countChar(char, massage):
    n = len(massage)
    count = 0
    for i in range(n):
        if massage[i] == char:
            count += 1
    return count


#count vowels

def countVowels(massage):
    n = len(massage)
    massage = massage.lower()
    count = 0
    for i in range(n):
        if (massage[i] == 'a' or massage[i] == 'e' or massage[i] == 'i' or massage[i] == 'o' or massage[i] == 'u'):
            count += 1
    return count

print(countChar('a', "thailand"))
print(countVowels("thailand"))